.github/workflows/build-and-test.yml

```diff
@@ -160,18 +161,18 @@ jobs:
           path: |
             ~/go/bin
             ~/go/pkg/mod
-          key: go-cache-${{ runner.os }}-${{ matrix.runner }}-${{ hashFiles('**/go.sum') }}
+          key: go-cache-${{ runner.os }}-${{ matrix.os }}-${{ hashFiles('**/go.sum') }}
       - name: Cache Build
         uses: actions/cache@1bd1e32a3bdc45362d1e726936510720a7c30a57 # v4.2.0
         with:
           path: ~/.cache/go-build
-          key: unittest-${{ runner.os }}-${{ matrix.runner }}-go-build-${{ matrix.go-version }}-${{ hashFiles('**/go.sum') }}
+          key: unittest-${{ runner.os }}-${{ matrix.os }}-go-build-${{ matrix.go-version }}-${{ hashFiles('**/go.sum') }}
       - name: Run Unit Tests
         run: |
           make -j4 gotest-with-junit
       - uses: actions/upload-artifact@v4
         with:
-          name: test-results-${{ runner.os }}-${{ matrix.runner }}-${{ matrix.go-version }}
+          name: test-results_${{ runner.os }}_${{ matrix.os }}_${{ runner.arch }}_${{ steps.setup-go.outputs.go-version }}
```

Comment:
wont this result in duplicated os name in the `name` string?

Reply:
Sort of - not quite.  Here's what the artifacts look like - 
 `test-results_Linux_ubuntu-latest_X64_1.22.10 `
so the runner.os is Linux, Macos, etc, and the matrix.os is what we use to request a runner - so ubuntu-latest in this particular example.  I captured both since technically one day we may test against different Linux distros

Reply:
I changed my mind.  If we ever want to do this we can - for now I eliminated the matrix.os variable for simplicity.  Plus the information is pretty duplicated and unspecific